File: src/main/java/rustleund/fightingfantasy/framework/closures/ClearPoisonDamageClosure.java
package rustleund.fightingfantasy.framework.closures;

import org.w3c.dom.Element;

import rustleund.fightingfantasy.framework.Closure;
import rustleund.fightingfantasy.framework.GameState;

public class ClearPoisonDamageClosure implements Closure {

	public ClearPoisonDamageClosure(@SuppressWarnings("unused") Element element) {
		// Nothing needed
	}

	@Override
	public boolean execute(GameState gameState) {
		gameState.getPlayerState().clearPoisonDamage();
		return true;
	}

}

File: src/main/java/rustleund/fightingfantasy/framework/closures/impl/ClearPoisonDamageClosure.kt
package rustleund.fightingfantasy.framework.closures.impl

import rustleund.fightingfantasy.framework.base.GameState
import rustleund.fightingfantasy.framework.closures.Closure

class ClearPoisonDamageClosure : Closure {

    override fun execute(gameState: GameState): Boolean {
        gameState.playerState.clearPoisonDamage()
        return true
    }
}